### PR TITLE
feat: check ssh and https url in bitbucketserver webhook #2773

### PR DESCRIPTION
**Pull request by eSamS**
Checklist:

* [x] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo-cd/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this does not need to be in the release notes.
* [x] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] I've updated both the CLI and UI to expose my feature, or I plan to submit a second PR with them.
* [ ] Optional. My organization is added to the README.
* [x] I've signed the CLA and my build is green ([troubleshooting builds](https://argoproj.github.io/argo-cd/developer-guide/ci/)).

### CODE REVIEW

**Review by jannfis — CHANGES_REQUESTED** [review]
Thanks for your PR, @eSamS ! Looks good so far, I just have some minor comments that I want to discuss. Please have a look.

File: util/webhook/webhook.go [hunk 1]
```diff
@@ -131,57 +133,62 @@ func affectedRevisionInfo(payloadIf interface{}) (string, string, bool) {
...
 
-	for _, app := range apps.Items {
-		if !repoRegexp.MatchString(app.Spec.Source.RepoURL) {
-			log.Debugf("%s does not match", app.Spec.Source.RepoURL)
-			continue
+	for _, webURL := range webURLs {
+		urlObj, err := url.Parse(webURL)
+		if err != nil {
+			log.Warnf("Failed to parse repoURL '%s'", webURL)
+			return
```
[the remark this answers is not in the document]
Comment (eSamS) [reply util/webhook/webhook.go]: You are right. I've added the changes.

File: util/webhook/webhook.go [hunk 2]
```diff
@@ -131,57 +133,62 @@ func affectedRevisionInfo(payloadIf interface{}) (string, string, bool) {
...
 		}
-		targetRev := app.Spec.Source.TargetRevision
-		if targetRev == "HEAD" || targetRev == "" {
-			if !touchedHead {
+		regexpStr := `(?i)(http://|https://|\w+@|ssh://(\w+@)?)` + urlObj.Host + "(:[0-9]+|)[:/]" + urlObj.Path[1:] + "(\\.git)?"
+		repoRegexp, err := regexp.Compile(regexpStr)
+		if err != nil {
+			log.Warn("Failed to compile repoURL regexp")
+			return
```
[the remark this answers is not in the document]
Comment (eSamS) [reply util/webhook/webhook.go]: Same here.

**Review by jannfis — APPROVED** [review]
LGTM, thank you!

**Pull request merged**